app/controllers/oai_controller.rb

```diff
@@ -92,16 +100,29 @@ def list_record_normalized_dump_candidates
 
       dumps = @stream.normalized_dumps.where(id: current_full_dump.id).or(current_full_dump.deltas).order(:created_at)
 
-      dumps = dumps.where(created_at: (Time.zone.parse(params[:from]).beginning_of_day)...) if params[:from]
-      dumps = dumps.where(created_at: ...(Time.zone.parse(params[:until]).end_of_day)) if params[:until]
-      dumps = dumps.where(id: params[:resumptionToken]...) if params[:resumptionToken]
+      if list_records_params[:from]
+        dumps = dumps.where(created_at: (Time.zone.parse(list_records_params[:from]).beginning_of_day)...)
+      end
+      dumps = dumps.where(created_at: ...(Time.zone.parse(list_records_params[:until]).end_of_day)) if list_records_params[:until]
+      dumps = dumps.where(id: list_records_params[:resumptionToken]...) if list_records_params[:resumptionToken]
 
       dumps
     end
   end
   # rubocop:enable Metrics/AbcSize
 
-  # rubocop:disable Metrics/AbcSize
+  # Wrap the provided Nokogiri::XML::Builder block in an OAI-PMH response
+  # See http://www.openarchives.org/OAI/openarchivesprotocol.html#XMLResponse
+  def build_oai_response(xml, params)
+    xml.send :'OAI-PMH', oai_xmlns do
+      xml.responseDate Time.zone.now.iso8601
+      xml.request(params.to_hash) do
+        xml.text oai_url
+      end
+      yield xml
+    end
+  end
+
   # rubocop:disable Metrics/MethodLength
   # See https://www.openarchives.org/OAI/openarchivesprotocol.html#ListRecords
   def build_list_records_response(dump, next_dump = nil)
```

Comment:
The `build_request` method is now gone, but it's being called from line 137 in the `build_oai_response` method.